### PR TITLE
Remove Currency class

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: src/core/orderbook.js [hunk 1]
```diff
@@ -1300,18 +1296,18 @@ OrderBook.prototype.toJSON =
 OrderBook.prototype.to_json = function() {
   const json = {
     taker_gets: {
-      currency: this._currencyGets.to_hex()
+      currency: this._currencyGets
```
Comment (darkdarkdragon) [inline src/core/orderbook.js]: This will change output. It is OK?

File: src/core/amount.js [hunk 2]
```diff
@@ -105,12 +106,12 @@ Amount.NaN = function() {
...
 };
 
-Amount.from_components_unsafe = function(value: Value, currency: Currency,
+Amount.from_components_unsafe = function(value: Value, currency: string,
   issuer: string, isNative: boolean
 ) {
   const result = new Amount(value);
   result._is_native = isNative;
-  result._currency = currency;
+  result._currency = normalizeCurrency(currency);
```
Comment (darkdarkdragon) [inline src/core/amount.js]: I think we should remove `normalizeCurrency` from here - it is caller's responsibility to provide correct values.

File: src/core/currency.js [hunk 3]
```diff
@@ -1,277 +1,58 @@
...
-
-  if (typeof j === 'number' && isFinite(j) && j >= 0) {
-    this._value = new BN(j);
-  }
-
-  this._update();
-  return this;
-};
+function isHexCurrency(currency) {
+  return /[A-Za-z0-9]{40}/.test(currency);
```
Comment (wltsmrz) [inline src/core/currency.js]: A-F?

File: src/core/currency.js [hunk 4]
```diff
@@ -1,277 +1,58 @@
...
-
-  // Json form:
-  //  '', 'XRP', '0': 0
-  //  3-letter code: ...
-  // XXX Should support hex, C++ doesn't currently allow it.
-
-  this._value = NaN;
-  this._update();
+function isISOCode(currency) {
+  return /^[A-Z0-9]{3}$/.test(currency);
```
Comment (wltsmrz) [inline src/core/currency.js]: https://github.com/ripple/rippled/blob/43a480f02dca3bc4740b13f202d1b1815a8f5415/src/ripple/protocol/impl/UintTypes.cpp#L31-L37

File: src/core/orderbook.js (hunk 1 above)
Comment (clark800) [inline src/core/orderbook.js]: I don't see any problem with it; I'm not sure why it was hex in the first place.

File: src/core/amount.js (hunk 2 above)
Comment (clark800) [inline src/core/amount.js]: Ok

File: src/core/currency.js (hunk 3 above)
Comment (clark800) [inline src/core/currency.js]: Oops! Yeah, will fix.

File: src/core/currency.js (hunk 4 above)
Comment (clark800) [inline src/core/currency.js]: Yeah, I remember that. Those aren't actually ISO codes though, so if we support them we would have to rename the function. My preference would be to just leave it as is and try to discourage people from using non-ISO 3-character codes.

File: src/core/orderbook.js (hunk 1 above)
Comment (darkdarkdragon) [inline src/core/orderbook.js]: Then OK

**Comment by clark800** [comment]
@darkdarkdragon @wltsmrz updated

**Comment by darkdarkdragon** [comment]
LGTM

**Pull request merged**